drivers/flash/soc_flash_silabs_series2.c

```diff
@@ -233,12 +304,24 @@ static DEVICE_API(flash, flash_silabs_driver_api) = {
 #endif
 };
 
+/* clang-format off */
 static struct flash_silabs_data flash_silabs_0_data = {
-#ifdef CONFIG_SOC_FLASH_SILABS_S2_DMA_WRITE
+#if defined(CONFIG_SOC_FLASH_SILABS_S2_DMA_WRITE) || defined(CONFIG_SOC_FLASH_SILABS_S2_DMA_READ)
 	.dma_dev = DEVICE_DT_GET(DT_INST_DMAS_CTLR(0)),
 	.dma_channel = -1,
 #endif
+#if defined(CONFIG_SOC_FLASH_SILABS_S2_DMA_READ)
+	.sync_status = -1,
```

Comment:
I believe you don't need to initialize this field (hence, you won't have to explain what `-1` means).

Reply:
I chose to defensively initialize this because it's set by the DMA complete ISR, but read from the thread. We want to catch bugs where the sync is skipped by initializing to an error value rather than 0 (success).

Reply:
Then I suggest `-EIO`.